### ISSUE TITLE
Only one service is served in Firebase serve

### ISSUE THREAD

**Issue by KeroppiMomo** [firebase/firebase-tools#1259]
### [REQUIRED] Environment info

**firebase-tools:**
Only 6.9.0 is affected. 6.8.0 is normal.

**Platform:**
OS: macOS
Node.js: 10.6.0
npm: 6.8.0

### [REQUIRED] Test case

Update `firebase-tools` to 6.9.0. Set up an environment with Hosting and Functions. Run `firebase serve` and only one service is set up.

### [REQUIRED] Steps to reproduce

My `firebase.json`: 
```
{
  "functions": {
    "predeploy": [
      "npm --prefix \"$RESOURCE_DIR\" run lint",
      "npm --prefix \"$RESOURCE_DIR\" run build"
    ],
    "source": "functions"
  },
  "hosting": {
    "public": "public",
    "rewrites": [{
        "source": "!__/**",
        "function": "redirects"
    }]
  },
  "firestore": {
    "rules": "firestore.rules",
    "indexes": "firestore.indexes.json"
  }
}
```

Run `npm install -g firebase-tools@6.9.0`

Set up a Firebase Hosting and Functions environment using `firebase init` (`hosting`/`functions`)

- For `firebase serve`, only hosting server is set up.
- For `firebase serve --only functions,hosting` and `firebase serve --only hosting,functions`, only functions server is set up:
```
✔  functions: Using node@10 from host.
✔  functions: Emulator running at http://localhost:5001
```

### [REQUIRED] Expected behavior

Start running the Hosting and Functions server.

### [REQUIRED] Actual behavior

Only set up one service:

#### With no options: only hosting is set up
```
$ firebase serve --debug
[2019-05-08T13:23:23.698Z] ----------------------------------------------------------------------
[2019-05-08T13:23:23.701Z] Command:       /usr/local/bin/node /usr/local/bin/firebase serve --debug
[2019-05-08T13:23:23.702Z] CLI Version:   6.9.0
[2019-05-08T13:23:23.702Z] Platform:      darwin
[2019-05-08T13:23:23.702Z] Node Version:  v10.6.0
[2019-05-08T13:23:23.703Z] Time:          Wed May 08 2019 21:23:23 GMT+0800 (GMT+08:00)
[2019-05-08T13:23:23.703Z] ----------------------------------------------------------------------

[2019-05-08T13:23:23.712Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[2019-05-08T13:23:23.712Z] > authorizing via signed-in user
[2019-05-08T13:23:23.712Z] [iam] checking project [project id] for permissions ["firebase.projects.get"]
[2019-05-08T13:23:23.714Z] >>> HTTP REQUEST POST https://cloudresourcemanager.googleapis.com/v1/projects/[project id]:testIamPermissions  
 permissions=[firebase.projects.get]
[2019-05-08T13:23:24.851Z] <<< HTTP RESPONSE 200 content-type=application/json; charset=UTF-8, vary=X-Origin, Referer, Origin,Accept-Encoding, date=Wed, 08 May 2019 13:23:24 GMT, server=ESF, cache-control=private, x-xss-protection=0, x-frame-options=SAMEORIGIN, x-content-type-options=nosniff, server-timing=gfet4t7; dur=953, alt-svc=quic=":443"; ma=2592000; v="46,44,43,39", accept-ranges=none, transfer-encoding=chunked
[2019-05-08T13:23:24.853Z] >>> HTTP REQUEST GET https://cloudresourcemanager.googleapis.com/v1/projects/[project id]
 
[2019-05-08T13:23:26.288Z] <<< HTTP RESPONSE 200 content-type=application/json; charset=UTF-8, vary=X-Origin, Referer, Origin,Accept-Encoding, date=Wed, 08 May 2019 13:23:25 GMT, server=ESF, cache-control=private, x-xss-protection=0, x-frame-options=SAMEORIGIN, x-content-type-options=nosniff, server-timing=gfet4t7; dur=1013, alt-svc=quic=":443"; ma=2592000; v="46,44,43,39", accept-ranges=none, transfer-encoding=chunked

=== Serving from '[path]'...

[2019-05-08T13:23:26.290Z] >>> HTTP REQUEST GET https://mobilesdk-pa.googleapis.com/v1/projects/575193212555/clients/_:getWebAppConfig  
 
[2019-05-08T13:23:27.098Z] <<< HTTP RESPONSE 200 content-type=application/json; charset=UTF-8, vary=X-Origin, Referer, Origin,Accept-Encoding, date=Wed, 08 May 2019 13:23:27 GMT, server=ESF, cache-control=private, x-xss-protection=0, x-frame-options=SAMEORIGIN, x-content-type-options=nosniff, alt-svc=quic=":443"; ma=2592000; v="46,44,43,39", accept-ranges=none, transfer-encoding=chunked
i  hosting: Serving hosting files from: public
✔  hosting: Local server: http://localhost:5000
```

#### With `--only functions,hosting` or `--only  hosting,functions`: only functions is set up
```
[2019-05-08T13:27:21.672Z] ----------------------------------------------------------------------
[2019-05-08T13:27:21.676Z] Command:       /usr/local/bin/node /usr/local/bin/firebase serve --only functions,hosting --debug
[2019-05-08T13:27:21.676Z] CLI Version:   6.9.0
[2019-05-08T13:27:21.676Z] Platform:      darwin
[2019-05-08T13:27:21.676Z] Node Version:  v10.6.0
[2019-05-08T13:27:21.677Z] Time:          Wed May 08 2019 21:27:21 GMT+0800 (GMT+08:00)
[2019-05-08T13:27:21.678Z] ----------------------------------------------------------------------

[2019-05-08T13:27:21.687Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[2019-05-08T13:27:21.688Z] > authorizing via signed-in user
[2019-05-08T13:27:21.688Z] [iam] checking project [project id] for permissions ["firebase.projects.get"]
[2019-05-08T13:27:21.690Z] >>> HTTP REQUEST POST https://cloudresourcemanager.googleapis.com/v1/projects/[project id]:testIamPermissions  
 permissions=[firebase.projects.get]
[2019-05-08T13:27:22.740Z] <<< HTTP RESPONSE 200 content-type=application/json; charset=UTF-8, vary=X-Origin, Referer, Origin,Accept-Encoding, date=Wed, 08 May 2019 13:27:22 GMT, server=ESF, cache-control=private, x-xss-protection=0, x-frame-options=SAMEORIGIN, x-content-type-options=nosniff, server-timing=gfet4t7; dur=962, alt-svc=quic=":443"; ma=2592000; v="46,44,43,39", accept-ranges=none, transfer-encoding=chunked
[2019-05-08T13:27:22.742Z] >>> HTTP REQUEST GET https://cloudresourcemanager.googleapis.com/v1/projects/[project id]
 
[2019-05-08T13:27:23.859Z] <<< HTTP RESPONSE 200 content-type=application/json; charset=UTF-8, vary=X-Origin, Referer, Origin,Accept-Encoding, date=Wed, 08 May 2019 13:27:23 GMT, server=ESF, cache-control=private, x-xss-protection=0, x-frame-options=SAMEORIGIN, x-content-type-options=nosniff, server-timing=gfet4t7; dur=1017, alt-svc=quic=":443"; ma=2592000; v="46,44,43,39", accept-ranges=none, transfer-encoding=chunked
✔  functions: Using node@10 from host.
[2019-05-08T13:27:23.869Z] >>> HTTP REQUEST GET https://mobilesdk-pa.googleapis.com/v1/projects/575193212555:getServerAppConfig  
 
[2019-05-08T13:27:24.214Z] <<< HTTP RESPONSE 200 content-type=application/json; charset=UTF-8, vary=X-Origin, Referer, Origin,Accept-Encoding, date=Wed, 08 May 2019 13:27:24 GMT, server=ESF, cache-control=private, x-xss-protection=0, x-frame-options=SAMEORIGIN, x-content-type-options=nosniff, alt-svc=quic=":443"; ma=2592000; v="46,44,43,39", accept-ranges=none, transfer-encoding=chunked
✔  functions: Emulator running at http://localhost:5001
```

**Comment by samtstern** [comment]
This is a duplicate of #1242, we will release the fix soon.  For now please downgrade to `6.8.0`

**Comment by samtstern** [comment]
The fix for this issue has been released in `6.9.1`.
